From 9ba16551970a079a0e89df63c127bd6110b9b375 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Fri, 7 Sep 2012 01:12:45 -0700
Subject: init.trace.rc: Key off of ro.debuggable, mount debugfs

Currently, if you want to use debugfs for tracing, you need to mount it
before the 'on boot' block in this file is processed, else the permissions
will be wrong. This means enabling debugfs is an all-or-nothing proposition,
it's not possible to do something in your init.<platform>.rc like:

    on property:ro.debuggable=1
        mount debugfs debugsfs /sys/kernel/debug

Since 'on property' triggers happen much later in the boot process.

The board configurations in AOSP all put the call to mount debugfs
in the early-init or boot blocks of theiir init.$(TARGET_PRODUCT).rc. This
is a security issue, debugfs should not be mounted in a production
configuration:

http://lwn.net/Articles/429328/

What this change does is to only enable debugfs and set the tracing
permissions in debug-enabled builds, and automatically mount debugfs
in those configurations.

Change-Id: I9e9734c2c453b07afbb6d01db80c591ef1bf816d
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 rootdir/init.trace.rc |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/rootdir/init.trace.rc b/rootdir/init.trace.rc
index 8a05fd0..4232474 100644
--- a/rootdir/init.trace.rc
+++ b/rootdir/init.trace.rc
@@ -1,6 +1,7 @@
 ## Permissions to allow system-wide tracing to the kernel trace buffer.
 ##
-on boot
+on property:ro.debuggable=1
+    mount debugfs debugfs /sys/kernel/debug
 
 # Allow writing to the kernel trace log.
     chmod 0222 /sys/kernel/debug/tracing/trace_marker
-- 
1.7.4.1

